lib/origen_testers/igxl_based_tester/ultraflex.rb

```diff
@@ -498,11 +496,213 @@ def store_next_cycle(*pins)
+      # issues the 'Start <waveform>' microcode
+      # Required arguments:
+      #                     pins
+      # Optional arguments:
+      #                     signal-name  (DigSrc waveform)
+      #                     starte   (true or false, defaults to false)
+      # Note restrestions from IG-XL help around use of starte opcode spacing to end of the current segment.
+      def digsrc_start(*pins)
+        options = pins.last.is_a?(Hash) ? pins.pop : {}
+        options = {
+          signal_name: '',
+          starte:      false
+        }.merge(options)
+
+        pins = pins.flatten.compact
+        options[:starte] ? start_syntax = 'Starte' : start_syntax = 'Start'
+        microcode "(#{format_multiple_instrument_pins(pins)}:DigSrc = #{start_syntax} #{options[:signal_name]})"
+
+        if options[:dssc_mode] == :single
+          $tester.cycle(repeat: 145) # minimum of 144 cycles, adding 1 for safey measures
+        elsif options[:dssc_mode] == :dual
+          $tester.cycle(repeat: 289) # minimum of 288 cycles, adding 1 for safety measures
+        else
+          $tester.cycle(repeat: 577) # minimum of 577 cycles, adding 1 for safety measures
+        end
+      end
+
+      # Call this method at the end of each digsrc overlay operation to clear the pattern
+      # memory pipeline, so that the pattern is ready to do the next digsrc overlay operation.
+      # Required arguments:
+      #                     pins
+      def digsrc_stop(*pins)
+        pins = pins.flatten.compact
+        microcode "(#{format_multiple_instrument_pins(pins)}:DigSrc = Stop)"
+      end
+
+      # Call this method to insert the digsrc SEND overlay microcode on the previous vector and change the pin state.
+      def digsrc_send(*pins)
+        dssc(pins, dssc_microcode: 'DigSrc = Send', pin_state: 'drive_mem')
+      end
+
+      # Call this method to insert the digsrc SEND overlay microcode on the next vector
+      def digsrc_send_next_cycle(*pins)
+        dssc_next_cycle(pins, dssc_microcode: 'DigSrc = Send', pin_state: 'drive_mem')
+      end
+      alias_method :digsrc_send!, :digsrc_send_next_cycle
+
+      # Call this method to insert the digcap TRIG microcode on the previous vector
+      def digcap_trig(*pins)
+        dssc(pins, dssc_microcode: 'DigCap = Trig')
+      end
+
+      # Call this method to insert the digcap TRIG microcode on the next vector
+      def digcap_trig_next_cycle(*pins)
+        dssc_next_cycle(pins, dssc_microcode: 'DigCap = Trig')
+      end
+      alias_method :digcap_trig!, :digcap_trig_next_cycle
+
+      # Call this method to insert the digcap STORE microcode on the previous vector
+      def digcap_store(*pins)
+        options = pins.last.is_a?(Hash) ? pins.pop : {}
+        options = {
+          offset:         0,
+          dssc_microcode: 'DigCap = Store'
+        }.merge(options)
+        dssc(pins, options)
+      end
```

Comment:
Should this not just be the store method?

Seems like dssc should be the default store method on UltraFLEX instead of STV?

Or more generally, allow the choice of approach to be selected by a tester attribute:

~~~ruby
my_reg.store!   # Uses dssc, assuming we make that the default here

tester.store_method = :stv

my_reg.store!   # Uses STV
~~~

Ultimately I think we want to integrate such an attribute with the tester configuration stuff that @scottchennxp added recently, so that it will automatically selects DSSC as the default capture method whenever the required instrumentation is present.